python/lsst/analysis/tools/actions/scalar/scalarActions.py

```diff
@@ -246,3 +246,45 @@ def __call__(self, data: KeyedData, **kwargs) -> Scalar:
         mask = self.getMask(**kwargs)
         arr = cast(Vector, data[self.vectorKey.format(**kwargs)])[mask]
         return cast(Scalar, np.nansum(arr))
+
+
+class MedianHistAction(ScalarAction):
+    """Calculates the median of the given histogram data."""
+
+    histKey = Field[str]("Key of frequency Vector to median")
+    midKey = Field[str]("Key of bin midpoints Vector to median")
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        return (
+            (self.histKey, Vector),
+            (self.midKey, Vector),
+        )
+
+    def histMedian(self, hist, bin_mid):
+        """Calculates the median of a histogram with binned values
+
+        Parameters
+        ----------
+        hist : `numpy.ndarray`
+            Frequency array
+        bin_mid : `numpy.ndarray`
+            Bin midpoints array
+
+        Returns
+        -------
+        median : `float`
+            Median of histogram with binned values
+        """
+        cumulative_sum = np.cumsum(hist)
+        median_index = np.searchsorted(cumulative_sum, cumulative_sum[-1] / 2)
+        median = bin_mid[median_index]
+        return median
+
+    def __call__(self, data: KeyedData, **kwargs) -> Scalar:
+        if len(data[self.histKey.format(**kwargs)]) != 0:
+            hist = cast(Vector, data[self.histKey.format(**kwargs)])
```

Comment:
You don't really need any casts here, unless you are thinking `histMedian` should be typed with `Vector` `Scalar` etc.